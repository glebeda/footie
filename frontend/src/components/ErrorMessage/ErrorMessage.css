.error-message {
    background-color: #f8d7da; 
    color: #721c24; 
    padding: 15px;
    margin: 20px 0;
    border: 1px solid #f5c6cb;
    border-radius: 4px;
  }
  