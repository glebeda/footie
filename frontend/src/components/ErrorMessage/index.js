import React from 'react';
import './ErrorMessage.css'; 

const ErrorMessage = ({ message }) => {
  return (
    <div className="error-message" role="alert">
      <p>{message}</p>
    </div>
  );
};

export default ErrorMessage;
