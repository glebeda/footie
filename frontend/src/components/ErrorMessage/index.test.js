import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorMessage from './index';

describe('ErrorMessage', () => {
  it('renders correctly with a message', () => {
    const message = 'An error occurred';
    const { asFragment } = render(<ErrorMessage message={message} />);
    expect(asFragment()).toMatchSnapshot();

    // Check that the message is displayed
    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it('has the correct role for accessibility', () => {
    render(<ErrorMessage message="Error!" />);
    expect(screen.getByRole('alert')).toBeInTheDocument();
  });
});
