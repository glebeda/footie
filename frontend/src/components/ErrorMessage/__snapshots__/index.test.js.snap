// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly with a message 1`] = `
<DocumentFragment>
  <div
    class="error-message"
    role="alert"
  >
    <p>
      An error occurred
    </p>
  </div>
</DocumentFragment>
`;
